File: Java/프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.java
import java.util.*;

class Solution {
    public int[] solution(int k, int[] score) {
        int[] answer = new int[score.length];
        PriorityQueue<Integer> pq = new PriorityQueue<>();
        for (int i = 0; i<score.length; i++) {
            int singerS = score[i];
            if (pq.size() < k) {
                pq.add(singerS);
            } else if (pq.peek() < singerS) {
                pq.poll();
                pq.add(singerS);
            }
            answer[i] = pq.peek();
        }
        return answer;
    }
}

File: Swift/프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.swift
import Foundation

func solution(_ k:Int, _ score:[Int]) -> [Int] {
    var answer: [Int] = []
    var list: [Int] = []
    for singer in score {
        if list.count < k {
            list.append(singer)
        } else if let first = list.first, first < singer {
            list.removeFirst()
            list.append(singer)
        }
        
        list.sort(by: { $0 < $1 })
        answer.append(list.first ?? 0)
    }
    return answer
}